protocol/validation/vmcontext.go

```diff
@@ -164,8 +110,66 @@ func NewTxVMContext(tx *bc.TxEntries, entry bc.Entry, prog bc.Program, args [][]
 		DestPos:       destPos,
 		AnchorID:      anchorID,
 		SpentOutputID: spentOutputID,
-		CheckOutput:   checkOutput,
+		CheckOutput:   (&txContext{entry: entry}).checkOutput,
 	}
 
 	return result
 }
+
+type txContext struct {
```

Comment:
I think I'd prefer a more specific name for this type such as `entryContext` or even `checkOutputContext`. "Tx" is a little overused, even allowing for the renaming simplification to come.